examples/plot_new_metrics.py

```diff
@@ -66,6 +67,8 @@
 from fairlearn.datasets import fetch_adult
 from fairlearn.metrics import MetricFrame, count, selection_rate
 
+pd.set_option("mode.copy_on_write", True)
```

Comment:
setting this option, basically sets the option for all examples being rendered after this one I think, since they are run in the same process. So we either want to unset at the end of the example, or we should be setting it in the configuration of `sphinx_gallery` or `sphinx` for all examples.

Reply:
Right, thanks. I would like to keep it local to the files for now, as setting this globally impacts the speed of the examples. Setting the `copy_on_write option` is the easiest blanket case to solve this without changing the code or its functionality, but there is another way to go around it using `df.iloc`, in which instead of the drop in place a subgroup of the df is kept. I will either unset or use the other option.

Reply:
I wanted to leave a "clean slate" for benchmarking with polars, but I can always turn it off for that. It seems it will be the default behaviour in the future versions, but it might come with some other improvements, I didn't read about it yet in detail. I think it makes sense though, I'll add it globally.

Reply:
Should we leave it in to allow this kind of behaviour in the future? As it is anyway becoming the default?